### PR TITLE
ISPN-9429 Ship default analyzers

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-9429

### CODE REVIEW

File: query/src/main/java/org/infinispan/query/impl/LifecycleManager.java [hunk 1]
```diff
@@ -321,7 +322,8 @@ private SearchIntegrator getSearchFactory(String cacheName, IndexingConfiguratio
       //defend against multiple initialization:
       if (searchFactory == null) {
          ClassLoader aggregatedClassLoader = makeAggregatedClassLoader(cr.getGlobalComponentRegistry().getGlobalConfiguration().classLoader());
-         Collection<ProgrammaticSearchMappingProvider> programmaticSearchMappingProviders = ServiceFinder.load(ProgrammaticSearchMappingProvider.class, aggregatedClassLoader);
+         Collection<ProgrammaticSearchMappingProvider> programmaticSearchMappingProviders = new HashSet<>(ServiceFinder.load(ProgrammaticSearchMappingProvider.class, aggregatedClassLoader));
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/impl/LifecycleManager.java]: Why add it manually and not use the regular mechanism for DefaultSearchMappingProvider ?

File: server/integration/testsuite/src/test/java/org/infinispan/server/test/query/RemoteQueryStringIT.java [hunk 2]
```diff
@@ -67,11 +59,10 @@
    @BeforeClass
    public static void before() {
       JavaArchive programmaticSearchMappingProviderArchive = ShrinkWrap.create(JavaArchive.class)
-            .addClass(TestAnalyzerProvider.class)
             .addClass(TestSearchMappingFactory.class)
             .addClass(TestSearchMappingFactory.MySearchableEntity.class)
             .add(new StringAsset("Dependencies: org.infinispan.query, org.hibernate.search.engine"), "META-INF/MANIFEST.MF")
-            .addAsServiceProvider(ProgrammaticSearchMappingProvider.class, TestAnalyzerProvider.class);
```
Comment (anistor) [inline server/integration/testsuite/src/test/java/org/infinispan/server/test/query/RemoteQueryStringIT.java]: This was also used to validate that TestAnalyzerProvider is indeed loadable in the very complex env of the server. I'd like to keep that test, even though now all those analyzers are preent by default. Let's still add one like that 'standard-with-stop' used in embedded test.

File: client/hotrod-client/src/test/resources/analyzers.proto [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    /* @Field(store = Store.YES, analyze = Analyze.YES, analyzer = @Analyzer(definition = "keyword")) */
+    optional string name4 = 4;
+
+    /* @Field(store = Store.YES, analyze = Analyze.YES, analyzer = @Analyzer(definition = "stemmer")) */
+    optional string name5 = 5;
+
+    /* @Field(store = Store.YES, analyze = Analyze.YES, analyzer = @Analyzer(definition = "ngram")) */
+    optional string name6 = 6;
+
+}
```
Comment (anistor) [inline client/hotrod-client/src/test/resources/analyzers.proto]: no EOL

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/testdomain/protobuf/TestEntity.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+         writer.writeString("name1", testEntity.name1);
+         writer.writeString("name2", testEntity.name2);
+         writer.writeString("name3", testEntity.name3);
+         writer.writeString("name4", testEntity.name4);
+         writer.writeString("name5", testEntity.name5);
+         writer.writeString("name6", testEntity.name6);
+      }
+
+      @Override
+      public Class<? extends TestEntity> getJavaClass() {
```
Comment (anistor) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/testdomain/protobuf/TestEntity.java]: ```public Class<TestEntity> getJavaClass() ``` is better :)

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/testdomain/protobuf/TestEntity.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      this.name4 = name4;
+      this.name5 = name5;
+      this.name6 = name6;
+   }
+
+   public static class TestEntityMarshaller implements MessageMarshaller<TestEntity> {
+
+      @Override
+      public TestEntity readFrom(ProtoStreamReader reader) throws IOException {
+         return new TestEntity(reader.readString("name1"), reader.readString("name2"),
```
Comment (anistor) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/testdomain/protobuf/TestEntity.java]: This lines makes a lot of pandas sad and depressed :)

File: server/integration/testsuite/src/test/java/org/infinispan/server/test/query/RemoteQueryStringIT.java (hunk 2 above)
Comment (gustavonalle) [reply server/integration/testsuite/src/test/java/org/infinispan/server/test/query/RemoteQueryStringIT.java]: The protobuf used in this test comes from the protostream codebase and hardcodes "standard", so this test needs to be rewritten, would you mind doing it later?

Comment (anistor) [reply server/integration/testsuite/src/test/java/org/infinispan/server/test/query/RemoteQueryStringIT.java]: No. So we need to change bank.proto model?

File: query/src/main/java/org/infinispan/query/impl/LifecycleManager.java (hunk 1 above)
Comment (gustavonalle) [reply query/src/main/java/org/infinispan/query/impl/LifecycleManager.java]: The final effect is the same, this makes it simpler and it works across all kinds of classloaders (osgi, wildfly, uberjar, etc)

File: server/integration/testsuite/src/test/java/org/infinispan/server/test/query/RemoteQueryStringIT.java (hunk 2 above)
Comment (gustavonalle) [reply server/integration/testsuite/src/test/java/org/infinispan/server/test/query/RemoteQueryStringIT.java]: I think it's better to leave it as is, as it's used throughout the code. I'd prefer to rewrite this test using more weird analyzers and different analyzers per field.

Comment (gustavonalle) [reply server/integration/testsuite/src/test/java/org/infinispan/server/test/query/RemoteQueryStringIT.java]: A good test would be to actually deploy a new analyzer like http://www.atilika.org/, and use it on one of the fields.

Comment (gustavonalle) [reply server/integration/testsuite/src/test/java/org/infinispan/server/test/query/RemoteQueryStringIT.java]: https://issues.jboss.org/browse/ISPN-9433

File: client/hotrod-client/src/test/resources/analyzers.proto (hunk 3 above)
Comment (anistor) [reply client/hotrod-client/src/test/resources/analyzers.proto]: Checkstyle only looks at java files

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/testdomain/protobuf/TestEntity.java (hunk 5 above)
Comment (anistor) [reply client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/testdomain/protobuf/TestEntity.java]: Each readString as a separate statement, so we do not give a bad example to others even if it just a test. But it's fine anyway :)

**Comment by anistor** [comment]
I keep getting OoM exceptions while running suite of client/hotrod-client

**Comment by gustavonalle** [comment]
What about ```master```?

**Comment by anistor** [comment]
Also on master. ClientEventsOOMTest fails, so I wonder what recent chnages lead to this.
```
[OK:  3281, KO:     1, SKIP:     0] Test failed: org.infinispan.client.hotrod.event.ClientEventsOOMTest.testOOM
java.lang.AssertionError: expected:<100> but was:<26>
	at org.testng.AssertJUnit.fail(AssertJUnit.java:59)
	at org.testng.AssertJUnit.failNotEquals(AssertJUnit.java:364)
	at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:80)
	at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:245)
	at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:252)
	at org.infinispan.client.hotrod.event.ClientEventsOOMTest.testOOM(ClientEventsOOMTest.java:94)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:196)
	at org.infinispan.commons.test.TestNGLongTestsHook.run(TestNGLongTestsHook.java:24)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:208)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:635)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.access$000(SuiteRunner.java:39)
	at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:393)
	at org.testng.internal.thread.ThreadUtil$2.call(ThreadUtil.java:64)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by anistor** [comment]
And after ClientEventsOOMTest fail, others random OoM failures follow.

**Comment by gustavonalle** [comment]
This test always pass for me locally

**Comment by gustavonalle** [comment]
I started to get some timeout in the hr client as well, it could be related to the recent threading changes in the Hot Rod server

**Comment by gustavonalle** [comment]
It seems to be the case. Each HotRodServer instance in those tests are now using dozens of netty IO threads rather than 3 fixed in the pom.xml a while ago https://github.com/infinispan/infinispan/pull/5221/files#diff-96fcd8ca614f0dc842b8166f0c11a96fR158

**Comment by anistor** [comment]
I'm very sure the changes in this PR are not causing these OoM exceptions. So I'll try to revert the HR threading changes locally and see if I can get a green.

**Comment by gustavonalle** [comment]
I can barely run the suite here locally since the suite forces ```-Xmx1G```, and I have 32Gb of memory :)
Do we have such a low value because of CI I suppose?

**Comment by gustavonalle** [comment]
@anistor Also, it appears some sort of leak was introduced recently, as profiling the testsuite I can see tons of ```org.infinispan.server.hotrod.test.HotRodTestingUtil$1``` objects accumulating in memory

**Comment by gustavonalle** [comment]
@anistor Please try with https://github.com/infinispan/infinispan/pull/6205, it's green to me

**Comment by anistor** [comment]
The other fixes are in now. Could you please rebase this @gustavonalle ?

**Comment by gustavonalle** [comment]
Rebased

**Comment by anistor** [comment]
LGTM. I'll squash the commits and merge it. Thanks @gustavonalle !

**Pull request closed without merging**